### PR TITLE
Fix direct compilation failed #1862

### PR DESCRIPTION
**Pull request by yangzhg**
Fix direct compilation failed：

1. fix compile thirdparty in ubuntu will install libs to `lib` dir instead of `lib64` 
2. fix compile error in gcc5 due to the defect of  c++11 (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=60970)
3. fix gcc version check will not work on some OS

### CODE REVIEW

File: be/src/olap/key_coder.h [hunk 1]
```diff
@@ -221,4 +221,13 @@ class KeyCoderTraits<OLAP_FIELD_TYPE_VARCHAR> {
     }
 };
 
+struct EnumClassHash
```
Comment (imay) [inline be/src/olap/key_coder.h]: 1. why not put this in cpp file?
2. we prefer `struct EnumClassHash {`

Comment (yangzhg) [reply be/src/olap/key_coder.h]: done

File: be/src/olap/key_coder.cpp [hunk 2]
```diff
@@ -27,6 +27,14 @@ KeyCoder::KeyCoder(TraitsType traits)
         _decode_ascending(traits.decode_ascending) {
 }
 
+struct EnumClassHash {
+    template <typename T>
+    std::size_t operator()(T t) const
```
Comment (imay) [inline be/src/olap/key_coder.cpp]:
```suggestion
    std::size_t operator()(T t) const {
```

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**